### PR TITLE
Release v1.6

### PR DESCRIPTION
**Pull request by francisco-dlp**
Update CHANGES for the release.

PRs with fixes and additions are more than welcome.

I'll direct the conda forge recipe here.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -45,9 +45,9 @@
 setup_path = os.path.dirname(__file__)
 
 
-install_req = ['scipy>=0.17',
+install_req = ['scipy>=1.4.0',
```
Comment (tjof2) [inline setup.py]: In which case, there is code in `model.py` and docs in `model.rst` which refer to LooseVersion of scipy >=1.2.0 for `dual_annealing` and `shgo` as options for `model.fit()`. This logic can be tidied up.

Comment (francisco-dlp) [reply setup.py]: Sounds good, would you like to take care of that in a PR to the main repository?

Comment (tjof2) [reply setup.py]: Yep I'll do it later this morning

Comment (francisco-dlp) [reply setup.py]: Great, thanks!

Comment (ericpre) [reply setup.py]: scipy 1.4 has been released in December 2019, which I think is a bit too recent. Scipy is not straightforward to package/distribute, so I would suggest to avoid increasing this requirement. As for numpy, in my opinion, it would be sensible to be conservative and as a guideline to supports scipy version released in the 2 years for example (see https://numpy.org/neps/nep-0029-deprecation_policy.html)

Comment (francisco-dlp) [reply setup.py]: The issue is that recent changes in ``decomposition`` require scipy v1.4.

Comment (ericpre) [reply setup.py]: Which changes exactly? There have been a few PRs releted to decomposition, recently!

Comment (francisco-dlp) [reply setup.py]: The [arpack test](https://github.com/hyperspy/hyperspy/blob/81c226ac4fc6a32e09022ce0ce2b29bb90308269/hyperspy/tests/learn/test_svd_pca.py#L82) fails for scipy < 1.4

Comment (tjof2) [reply setup.py]: Note that 1.4 also brings `scipy.fft` over `fftpack`. You can do a `pytest.importorskip("scipy", minversion="1.4")` for that test.

Comment (tjof2) [reply setup.py]: It can definitely go to scipy 1.0.1 because that's what scikit-image requires (cf. #1438).

I won't make a separate PR in this case given @ericpre's comment.

Comment (francisco-dlp) [reply setup.py]: The thing is that, AFAIK, hyperspy is not available in any Linux distribution other than the ones with a rolling release policy such as Arch and [OpenSuse Tumbleweed](https://build.opensuse.org/package/show/home:andythe_great/python-hyperspy), for which requiring a recent scipy is not an issue. That's certainly because our dependencies, even when being conservative regarding scipy and numpy as currently, cannot be fulfilled in other distributions. Hopefully the situation will change with "The Split", but, until then, maybe it is not too bad to depend on a recent scipy?

Comment (ericpre) [reply setup.py]: It seems that `svd_pca` will fail with `decomposition_output=None` and `svd_solver='arpack'.` but when using `decomposition`, the `output_dimension` is set (not None), so it should be fine to skip the failing test for this specific case?

File: CHANGES.rst [hunk 2]
```diff
@@ -84,6 +84,33 @@ Enhancements
...
+    * "fmin_l_bfgs_b" -> "L-BFGS-B",
+    * "fmin_tnc" -> "TNC",
+    * "fmin_powell" -> "Powell",
+    * "mpfit" -> "lm" (in combination with `"bounded=True"`),
+    * "leastsq" -> "lm",
+    * Passing integer arguments to ``parallel`` to select the number of
+      workers is now deprecated. Use ``parallel=True, max_workers={value}``
+      instead.
+* The ``method`` keyword has been renamed to ``loss_function``.
+* The ``loss_funtion`` value ``"ml"`` has been renamed to ``ml-poisson``.
```
Comment (tjof2) [inline CHANGES.rst]: Typo in loss function

File: setup.py (hunk 1 above)
Comment (tjof2) [reply setup.py]: No, the test for None is actually necessary, see here: https://github.com/hyperspy/hyperspy/blob/3ec3018f59cee4c9db023408b09d51024fe53d42/hyperspy/learn/svd_pca.py#L128-L131

File: CHANGES.rst (hunk 2 above)
Comment (francisco-dlp) [reply CHANGES.rst]: Fixed, thanks. Could you have a look at it again? I should be fairly complete now.

File: CHANGES.rst [hunk 3]
```diff
@@ -1,11 +1,176 @@
...
+  is deprecated and will be removed.
+* Several syntax changes in the :py:meth:`~.learn.mva.decomposition` method:
+
+  * Several ``algorithm`` keyword values have been renamed as follows:
+
+    * ``"fast_svd"``: ``"svd"``,
+    * ``"fast_mlpca"``: ``"mlpca"``,
+    * ``"RPCA_GoDec"``: ``"rpca"``,
+    * ``"ORPCA"``: ``"orpca"``,
+    * ``"ORNMF"``: ``"ornmf"``,
```
Comment (tjof2) [inline CHANGES.rst]: these need to be uppercase after #2467

**Review by tjof2 — APPROVED** [review]

File: CHANGES.rst (hunk 2 above)
Comment (tjof2) [reply CHANGES.rst]: Have done so

Comment (francisco-dlp) [reply CHANGES.rst]: Thanks!

What do you think about renaming:

* "ml-poisson" -> "ML-Poisson" or "ML-poisson"?
* "lm" -> "LM"

to make it clearer that they're acronyms?

Comment (tjof2) [reply CHANGES.rst]: - I think keeping `"lm"` lowercase is important to match `scipy.optimize.least_squares`.
- Happy with `"ML-poisson"`

**Comment by francisco-dlp** [comment]
Test release at https://test.pypi.org/project/hyperspy/1.6.0/

**Comment by francisco-dlp** [comment]
@ericpre, I think that this is v1.6, do you agree?

**Comment by ericpre** [comment]
I am finishing checking that everything is fine with the GUIs - I am not 100% sure if a few more tweaks are necessary in hyperspy.

**Comment by ericpre** [comment]
#2470

File: setup.py (hunk 1 above)
Comment (ericpre) [reply setup.py]: I still think that scipy 1.4 is too recent. I will make a PR to raise an error when using arpack svd solver and scipy < 1.4.

Comment (francisco-dlp) [reply setup.py]: OK, we can go down to v1.1 that is 
already in Debian.

Comment (ericpre) [reply setup.py]: #2471

Comment (francisco-dlp) [reply setup.py]: OK, I'll set scipy==1.0.1 as suggested by @tjof2

**Review by ericpre — APPROVED** [review]

**Comment by ericpre** [comment]
I have just realised that the wheels are not being build for python 3.7 on win and mac... We could do this on azure pipeline with these build running "on tag".

**Comment by francisco-dlp** [comment]
Yes, I've just realized that myself too. However, I think that we can get them from azure's ariefacts isn't it?

**Comment by ericpre** [comment]
yes but with the current setup, we are missing two wheels (python 3.7 for mac and windows):
https://dev.azure.com/franciscode-la-pena-manchon/hyperspy/_build/results?buildId=536&view=artifacts&type=publishedArtifacts

I will see if this is possible to run a pipeline on azure for the two missing wheel and without making a new tag...

**Comment by francisco-dlp** [comment]
Hopefully the tag won't matter as long as  we  don't change the version in the package. So, it should be enough to add the extra builds to azure without changing anything just to get the file from there. I may be wrong of course.

Could you upload the wheels to pypi? I should help packaging other sort stuff now... The widgets are now released in pypi.

**Comment by ericpre** [comment]
Yes, I will take the commit from which you made the tag to run the to run the pipeline and then I will upload the wheels on pypi!

**Comment by francisco-dlp** [comment]
You could take the following one in which I add update the DOI, that's the one that I've used for the sdist.

**Comment by ericpre** [comment]
ah yes and this is the one which is on pypi, so it is better to take that one!
